iceoryx_examples/icediscovery/iox_find_service.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/posix_wrapper/signal_handler.hpp"
+//! [include poshdiscovery]
+#include "iceoryx_posh/runtime/posh_discovery.hpp"
+//! [include poshdiscovery]
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+
+#include <iostream>
+
+bool killswitch = false;
+constexpr char APP_NAME[] = "iox-cpp-find-service";
+
+static void sigHandler(int f_sig IOX_MAYBE_UNUSED)
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    iox::runtime::PoshDiscovery poshDiscovery;
+
+    // run until interrupted by Ctrl-C
+    while (!killswitch)
+    {
+        /// @todo #415 Instead of polling should we showcase the getServiceRegistryChangeCounter() or will that be
+        /// removed?
+
+        /// @todo #415 Why is the implicit conversion to cxx::variant not working?
```

Comment:
I remember that you wanted to change something in the variant to make this possible

Reply:
Yep and I thought I changed that already, maybe it was a just dream :grin: I'll double check.

Reply:
@elBoberido Found my [previous answer](https://github.com/eclipse-iceoryx/iceoryx/pull/864/files#r658051675).

Reply:
Maybe we have to rethink the approach with the variant. Could be something like wrapping it into a `newtype` with ctors for `IdString` and `Wildcard_t`